src/main/diagnostics/descriptorActionProvider.ts

```diff
@@ -21,52 +23,99 @@ export class DescriptorActionProvider extends AbstractFileActionProvider {
         // Search if the descriptor had issues in the scan
         const fileIssues: FileTreeNode | undefined = this._treesManager.issuesTreeDataProvider.getFileIssuesTree(document.uri.fsPath);
         if (fileIssues instanceof DescriptorTreeNode) {
-            this._treesManager.logManager.logMessage("Creating diagnostics for document '" + document.uri.fsPath + "'", 'DEBUG');
+            this._treesManager.logManager.logMessage("Creating diagnostics for descriptor '" + document.uri.fsPath + "'", 'DEBUG');
             const textEditor: vscode.TextEditor = await vscode.window.showTextDocument(document);
             let diagnostics: vscode.Diagnostic[] = [];
-            let proceesedDependencies: Set<string> = new Set<string>();
+            let proceesedDependencies: Map<string, InfoPair> = new Map<string, InfoPair>();
             // Get the direct dependency of each issue in the descriptor from the impact tree
             fileIssues.dependenciesWithIssue.forEach(dependencyWithIssue => {
                 dependencyWithIssue.issues.forEach(issue => {
                     issue.impactedTree?.children
                         ?.map(impact => impact.name)
-                        .forEach(directDependencyId => {
-                            if (proceesedDependencies.has(directDependencyId)) {
-                                // Direct dependency already had a diffrent dependency with issue that generated information for it
-                                return;
-                            }
-                            // Create diagnostics for the dependency
-                            proceesedDependencies.add(directDependencyId);
-                            this.generateInformation(
+                        .forEach(directDependencyId =>
```

Comment:
Why do we need to iterate twice here?

Reply:
we iterate only once.
for each file it has a list of dependencies with issues we iterate for each dependency.
on each dependency there is a list of issues so we iterate for each issue.
on each issue we have its impact path, from this path we take the first depth children those are the direct dependency that are infected with this issue and we pass them (the issue and the direct dependency) to be handled in the method